src/OrchardCore.Modules/OrchardCore.Deployment.Remote/Controllers/ImportRemoteInstanceController.cs

```diff
@@ -40,13 +50,27 @@ public async Task<IActionResult> Import(ImportViewModel model)
             var remoteClientList = await _remoteClientService.GetRemoteClientListAsync();
 
             var remoteClient = remoteClientList.RemoteClients.FirstOrDefault(x => x.ClientName == model.ClientName);
-
             if (remoteClient == null)
             {
                 return StatusCode((int)HttpStatusCode.BadRequest, "The remote client was not provided");
             }
 
-            var apiKey = Encoding.UTF8.GetString(_dataProtector.Unprotect(remoteClient.ProtectedApiKey));
+            var apiKey = string.Empty;
+            if (!string.IsNullOrEmpty(remoteClient.ApiKeySecret))
+            {
+                apiKey = (await _secretService.GetSecretAsync<TextSecret>(remoteClient.ApiKeySecret))?.Text;
+            }
+            else
+            {
+                try
+                {
+                    apiKey = Encoding.UTF8.GetString(_dataProtector.Unprotect(remoteClient.ProtectedApiKey));
+                }
+                catch
+                {
+                    _logger.LogError("The Api Key could not be decrypted. It may have been encrypted using a different key.");
+                }
+            }
```

Comment:
Is this for backward compatibility? If yes, then please comment that. Also, I think even on read, the new value, i.e. `ApiKeySecret`, should be written, so gradually data is migrated to the new format. And the same in all similar cases.

Reply:
Okay I will add comments.

Yes we support both a secret and a plain text but on the different UI when a secret is selected we no longer see the plain text one, unless if we select `None` again in the secrets list.

Reply:
How about, while keeping this backward compatibility for existing instances, we'd automatically store such passwords as secrets? I.e., you could still provide an SMTP password on the Email settings page, but from now on, it would be saved as an automatically created secret.

Because the UX now seems a little bit disconnected for me for such use cases: you have to separately create a secret under a separate feature, and then come back and select it. I'm not sure people will make the effort all the time (since it's just easier to paste the password into the field), without understanding the consequences.

Reply:
Yes I understand and I thought about it but it was not so obvious to define the strategy.

So maybe I was too lazy ;)

- So if we go this way I think we need an auto migration to secrets.

- Hmm, what would be the name of these auto created secrets?

- Then yes we need to provide the ability to create / edit a secret from the related editors, for example to rename an auto created secret.

I thought it was too risky but yes looks like it is the right way, interesting but more work to do.

Reply:
Yeah, I think so. These would be like feature-managed secrets. Each feature can name it according to their IDs, like `OrchardCore.Email.SmtpSettings.Password`.

Reply:
Related: https://github.com/OrchardCMS/OrchardCore/pull/7891#discussion_r1380585731